built-in-services/account/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+use crate::types::{
+    Account, GenerateAccountPayload, GenerateAccountResponse, GetAccountPayload, PayloadAccount,
+    Permission, VerifyPayload, VerifyResponse, Witness, ACCOUNT_TYPE_MULTI_SIG,
+    ACCOUNT_TYPE_PUBLIC_KEY, MAX_PERMISSION_ACCOUNTS,
+};
+use binding_macro::{cycles, service};
+use bytes::Bytes;
+use common_crypto::{Crypto, Secp256k1};
+use hasher::{Hasher, HasherKeccak};
+use protocol::traits::{ExecutorParams, ServiceResponse, ServiceSDK};
+use protocol::types::{Address, Hash, Hex, ServiceContext};
+
+#[cfg(test)]
+mod tests;
+pub mod types;
+
+pub struct AccountService<SDK> {
+    sdk: SDK,
+}
+
+#[service]
+impl<SDK: ServiceSDK> AccountService<SDK> {
+    pub fn new(sdk: SDK) -> Self {
+        Self { sdk }
+    }
+
+    #[cycles(100_00)]
+    #[read]
+    fn verify_signature(
+        &self,
+        ctx: ServiceContext,
+        payload: VerifyPayload,
+    ) -> ServiceResponse<VerifyResponse> {
+        let wit_str = String::from_utf8(payload.witness.to_vec());
+        if wit_str.is_err() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                112,
+                "witness from utf8 bytes to string failed".to_owned(),
+            );
+        }
+
+        let wit_res: Result<Witness, _> = serde_json::from_str(wit_str.unwrap().as_str());
+        if wit_res.is_err() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                112,
+                "witness not valid".to_owned(),
+            );
+        }
+
+        let wit = wit_res.unwrap();
+        if wit.signature_type == ACCOUNT_TYPE_PUBLIC_KEY {
+            if wit.signatures.len() != 1 || wit.pubkeys.len() != 1 {
+                return ServiceResponse::<VerifyResponse>::from_error(
+                    113,
+                    "len of signatures or pubkeys must be 1".to_owned(),
+                );
+            }
+            return verify_single_sig(&payload.tx_hash, &wit.signatures[0], &wit.pubkeys[0]);
+        }
+
+        // multi sig
+        if wit.signature_type != ACCOUNT_TYPE_MULTI_SIG {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                114,
+                "signature_type not valid".to_owned(),
+            );
+        }
+
+        let permission = self
+            .sdk
+            .get_account_value(&wit.sender, &0u8)
+            .unwrap_or(Permission {
+                accounts:  Vec::<Account>::new(),
+                threshold: 0,
+            });
+
+        if permission.threshold == 0 {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                110,
+                "account not existed".to_owned(),
+            );
+        }
```

Comment:
Why use `unwrap_or` and check for `threshold`? IMO, check whether the result of `get_account_value` is `None` or `Some` is better.

Reply:
actually , check for threshold  is the alternative coding for None check.
I think your advise is right, i will rewrite it your way